packages/launchpad/src/setup-actions.ts

```diff
@@ -0,0 +1,56 @@
+import { isMobile } from '@automattic/viewport';
+import { addQueryArgs } from '@wordpress/url';
+import type { Task } from './types';
+
+export const setUpActions = ( tasks: Task[], siteSlug: string | null ): Task[] => {
+	// Add actions to known tasks
+	return tasks.map( ( task: Task ) => {
+		let actionDispatch;
+
+		switch ( task.id ) {
+			case 'site_title':
+				actionDispatch = () => {
+					window.location.assign( `/settings/general/${ siteSlug }` );
+				};
+				break;
+
+			case 'design_edited':
+				actionDispatch = () => {
+					window.location.assign(
+						addQueryArgs( `/site-editor/${ siteSlug }`, {
+							canvas: 'edit',
+						} )
+					);
+				};
+				break;
+
+			case 'domain_claim':
+			case 'domain_upsell':
+			case 'domain_customize':
+				actionDispatch = () => {
+					window.location.assign( `/domains/add/${ siteSlug }` );
+				};
+				break;
+			case 'drive_traffic':
+				actionDispatch = () => {
+					const url = isMobile()
+						? `/marketing/connections/${ siteSlug }`
+						: `/marketing/connections/${ siteSlug }?tour=marketingConnectionsTour`;
+					window.location.assign( url );
+				};
+				break;
+			case 'add_new_page':
+				actionDispatch = () => {
+					window.location.assign( `/page/${ siteSlug }` );
+				};
+				break;
+			case 'edit_page':
+				actionDispatch = () => {
+					window.location.assign( `/pages/${ siteSlug }` );
+				};
+				break;
+		}
+
+		return { ...task, actionDispatch };
+	} );
+};
```

Comment:
To be a bit more explicit, I think it may be better to call this something like `setUpActionsForTasks()`, and I think it may actually be worth considering some additional arguments to the function to push more code into a central location so we can reduce code duplication further. Maybe we can look at splitting this into multiple diffs, but I think it could also be worth tackling it all in this PR.
```suggestion
type ExtraTaskProps = {
	recordTaskClickTracksEvent: ( task: Task ) => void;
	processActionForTask?: ( task: Task, action: undefined | () => void ): undefined | () => void;
}
export const setUpTaskActions = ( tasks: Task[], siteSlug: string | null, extraTaskProps: ExtraTaskProps ): Task[] => {
	const { recordTaskClickTracksEvent, processActionForTask } = extraTaskProps;

	// Add actions to known tasks
	return tasks.map( ( task: Task ) => {
		let action;
		
		switch ( task.id ) {
			case 'site_title':
				action = () => {
					window.location.assign( `/settings/general/${ siteSlug }` );
				};
				break;

			case 'design_edited':
				action = () => {
					window.location.assign(
						addQueryArgs( `/site-editor/${ siteSlug }`, {
							canvas: 'edit',
						} )
					);
				};
				break;

			case 'domain_claim':
			case 'domain_upsell':
			case 'domain_customize':
				action = () => {
					window.location.assign( `/domains/add/${ siteSlug }` );
				};
				break;
			case 'drive_traffic':
				action = () => {
					const url = isMobile()
						? `/marketing/connections/${ siteSlug }`
						: `/marketing/connections/${ siteSlug }?tour=marketingConnectionsTour`;
					window.location.assign( url );
				};
				break;
			case 'add_new_page':
				action = () => {
					window.location.assign( `/page/${ siteSlug }` );
				};
				break;
			case 'edit_page':
				action = () => {
					window.location.assign( `/pages/${ siteSlug }` );
				};
				break;
		}

		if ( processActionForTask ) {
			action = processActionForTask( task, action );
		}

		const actionDispatch= () => {
			// Note that task won't have actionDispatch defined but I think that's OK for tracking purposes.
			recordTaskClickTracksEvent( task );
			action?.();
		};

		return { ...task, actionDispatch };
	} );
};
```

Reply:
@daledupreez I took your idea and made some modifications, consolidating the Tracks view and click events into the new actions file to DRY things up.